contrib/python/convection-box.prm

```diff
@@ -0,0 +1,165 @@
+# A description of convection in a 2d box. See the manual for more information.
```

Comment:
Is this just a copy of the cookbook? Could you then instead reference the original .prm file in the python notebook instead of making a copy? This way we avoid the duplication. Duplicated files can always create problems if they get out of sync.

If it is just nearly a copy with some changes, you can do the following: Remove everything below the top comment, and then put in:

```
set Dimension = 2

include $ASPECT_SOURCE_DIR/cookbooks/convection-box.prm

set all parameters that have been changed from the original cookbook here, below the include statement
```

This uses the original cookbook as a base model, and then overwrites any parameters you want to change.

Reply:
Oh, I did not know this trick! I'll do that.

Reply:
Do we need to keep the set Dimension = 2 for a specific reason? Or can I only include the prm file?

Reply:
Yes you need to keep the Dimension parameter, because it is needed before the `include` directive is resolved (the current version should not run).

Reply:
The current version is running correctly... So I'm surprised, it seems to be OK without including the Dimension in the prm file. Could you test?

Reply:
You are right, sorry for the confusion, I misremembered that. The problem is as I described, but if ASPECT does not find the `Dimension` parameter in such a file, it simply assumes 2 (i.e. if that were a 3D model it would have crashed). Then this is good to go.